### DISCUSSION TITLE
How to use FilteredStream

### DISCUSSION THREAD

**Discussion by santeiro** [redouane59/twittered#443] [Q&A] [answered]
Hi everyone,

I am new to twittered and can't find any documentation on how to use the filtered stream.

The only example I could find is:
`Future<Response> future = twitterClient.startFilteredStream(System.out::println);`

A sample code snippet on how to get the response, print the text of the tweet would be great.

Thank you
Roberto
ps: will try to figure out on my own how to set up the filter

**Comment by redouane59** [comment, marked as answer]
Hello, here is an example : 

```
public void start() {

    try {
      this.streamResponse = TWITTER_CLIENT.startFilteredStream(new IAPIEventListener() {
        @SneakyThrows
        @Override
        public void onStreamError(final int httpCode, final String error) {
          // ...
        }

        @Override
        public void onTweetStreamed(final Tweet tweet) {
             // ...
        }

        @Override
        public void onUnknownDataStreamed(final String json) {
          // ...
        }

        @Override
        public void onStreamEnded(final Exception e) {
          // ...
        }
      });
    } catch (Exception e) {
      System.err.println("- error - " + e.getMessage());
    }
  }
```

**Reply by santeiro** [reply]
Thank you @redouane59 

resolved adding this to `public static void main`

`//public void start() {

        try {
          twitterClient.startFilteredStream(new IAPIEventListener() {
              /*@SneakyThrows*/
            @Override
            public void onStreamError(final int httpCode, final String error) {
              // ...
            }

            @Override
            public void onTweetStreamed(final Tweet tweet) {
                 System.out.println(tweet.getText());
            }

            @Override
            public void onUnknownDataStreamed(final String json) {
              // ...
            }

            @Override
            public void onStreamEnded(final Exception e) {
              // ...
            }
          });
        } catch (Exception e) {
          System.err.println("- error - " + e.getMessage());
        }`

Need to figure out how to use the void start method, and `@SneakyThrows` was giving me errors

Also getting this error while running the filtered stream - the stream works correctly after the exception is reported
`SLF4J: Class path contains multiple SLF4J providers.
SLF4J: Found provider [ch.qos.logback.classic.spi.LogbackServiceProvider@6842775d]
SLF4J: Found provider [org.slf4j.simple.SimpleServiceProvider@574caa3f]
SLF4J: See https://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual provider is of type [ch.qos.logback.classic.spi.LogbackServiceProvider@6842775d]
Unexpected problem occured during version sanity check
Reported exception:
java.lang.AbstractMethodError: Receiver class ch.qos.logback.classic.spi.LogbackServiceProvider does not define or inherit an implementation of the resolved method 'abstract java.lang.String getRequestedApiVersion()' of interface org.slf4j.spi.SLF4JServiceProvider.
	at org.slf4j.LoggerFactory.versionSanityCheck(LoggerFactory.java:331)
	at org.slf4j.LoggerFactory.performInitialization(LoggerFactory.java:172)
	at org.slf4j.LoggerFactory.getProvider(LoggerFactory.java:455)
	at org.slf4j.LoggerFactory.getILoggerFactory(LoggerFactory.java:441)
	at org.slf4j.LoggerFactory.getLogger(LoggerFactory.java:390)
	at org.slf4j.LoggerFactory.getLogger(LoggerFactory.java:416)
	at io.github.redouane59.twitter.TwitterClient.<clinit>(TwitterClient.java:100)
	at com.santeiro.twitterV2.Twittered.FilteredStream.FilteredStream.main(FilteredStream.java:37)`

**Comment by redouane59** [comment]
`@SneakyThrows` comes from lombok but you can use classic try/catch if you prefer.
Then you have to add some rules to tell to the API what you are listening : https://developer.twitter.com/en/docs/twitter-api/tweets/filtered-stream/integrate/build-a-rule
